easybuild/easyconfigs/g/GDRCopy/GDRCopy-2.1-GCCcore-9.3.0-CUDA-11.0.2.eb

```diff
@@ -0,0 +1,46 @@
+easyblock = 'ConfigureMake'
+
+name = 'GDRCopy'
+version = '2.1'
+local_cudaversion = '11.0.2'
+versionsuffix = '-CUDA-%s' % local_cudaversion
+
+homepage = 'https://github.com/NVIDIA/gdrcopy'
+description = "A low-latency GPU memory copy library based on NVIDIA GPUDirect RDMA technology."
+
+toolchain = {'name': 'GCCcore', 'version': '9.3.0'}
+toolchainopts = {'pic': True}
+
+github_account = 'NVIDIA'
+source_urls = [GITHUB_SOURCE]
+sources = ['%(version)s.tar.gz']
+checksums = ['cecc7dcc071107f77396f5553c9109790b6d2298ae29eb2dbbdd52b2a213e4ea']
+
+builddependencies = [
+    ('binutils', '2.34'),
+    ('Autotools', '20180311'),
+    ('pkg-config', '0.29.2'),
+]
+
+osdependencies = [
+    ('gdrcopy-kmod', 'gdrdrv-dkms'),  # kernel module for GDR Copy
+    'nvidia_peer_memory',  # kernel module for GPUDirect RDMA
```

Comment:
These extra OS dependencies are going to make the installation significantly more painful, but I guess there's no way around this...

Until now I could test the installation of a lot of the `fosscuda` easyconfigs on a non-GPU system, but that'll break with this in place.

That's not a good reason not to do this though, of course. :)

Reply:
Yeah I agree, we recently changed our deployment scripts with EB to ensure that all stuff using CUDA is built in a GPU node. Otherwise start to break more and more.
BTW, each of this kernel modules comes with its own system service that needs to be enabled. Just to make thing even more convoluted :smile:

Reply:
So, gdrcopy-kmod/gdrdrv-dkms, do you guys build your own RPMs for that? I'm a bit surprised not to find them in the cuda-repos or anything.

Reply:
I'm a little worried about the OS deps too. Moreover, it looks like these are run-time dependencies rather than compile-time dependencies.

Reply:
And since the "module load" doesn't check for RPMs I'm not sure what the logic is for obliging their existence at compile time. We should check if
a) GDRCopy still builds without the os deps
b) The whole setup is still usable without the kernel module RPMs, just less functional.

Reply:
They're required at build time too, I ran into trouble on a non-GPU node (missing headers, I think).

Reply:
Mmmm, we have been building CUDA stuff on a non-GPU node forever. We only have the CUDA libraries (i.e. the `.so`'s). Not the drivers, nor the hardware.

Reply:
Like this:

```
make[2]: Entering directory `/usr/src/kernels/3.10.0-1127.8.2.el7.ug.x86_64'
find: /usr/src/nvidia-*: No such file or directory
dirname: missing operand
Try 'dirname --help' for more information.
  CC [M]  /tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv/nv-p2p-dummy.o
  CC [M]  /tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv/gdrdrv.o
/tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv/nv-p2p-dummy.c:48:10: fatal error: nv-p2p.h: No such file or directory
   48 | #include "nv-p2p.h"
      |          ^~~~~~~~~~
compilation terminated.                                                                                                                                                                                     /tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv/gdrdrv.c:121:10: fatal error: nv-p2p.h: No such file or directory
  121 | #include "nv-p2p.h"
      |          ^~~~~~~~~~
compilation terminated.
make[3]: *** [/tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv/nv-p2p-dummy.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [/tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv/gdrdrv.o] Error 1
make[2]: *** [_module_/tmp/vsc40023/easybuild_build/GDRCopy/2.1/GCCcore-9.3.0-CUDA-11.0.2/gdrcopy-2.1/src/gdrdrv] Error 2
make[2]: Leaving directory `/usr/src/kernels/3.10.0-1127.8.2.el7.ug.x86_64'
```

Reply:
Yeah, that was working fine before, but no longer when GDRCopy is involved...

Reply:
@boegel does it compile for you if you 
`buildopts = "config lib exes %s" % local_envopts`
otherwise it tries to compile a kernel module but that's outside the scope of easybuild...

Reply:
@Micket we build those RPMs ourselves from their github repositories